dev/PushNotifications/PushNotificationManager.h

```diff
@@ -4,22 +4,49 @@
 #pragma once
 #include "Microsoft.Windows.PushNotifications.PushNotificationManager.g.h"
 #include <windows.foundation.h>
+#include <NotificationsLongRunningProcess_h.h>
+#include <frameworkudk/pushnotificationsRT.h>
 
 namespace winrt::Microsoft::Windows::PushNotifications::implementation
 {
+    typedef winrt::Windows::Foundation::TypedEventHandler<
+        winrt::Microsoft::Windows::PushNotifications::PushNotificationManager,
+        winrt::Microsoft::Windows::PushNotifications::PushNotificationReceivedEventArgs> PushNotificationEventHandler;
 
-    struct PushNotificationManager : PushNotificationManagerT<PushNotificationManager>
+    struct PushNotificationManager : PushNotificationManagerT<PushNotificationManager, IWpnForegroundSink, ABI::Microsoft::Internal::PushNotifications::INotificationListener>
     {
-        PushNotificationManager() = default;
+        PushNotificationManager();
 
         static winrt::Microsoft::Windows::PushNotifications::PushNotificationManager Default();
```

Comment:
Be mindful when putting COM / WinRT types into static variables. Unless you control for it, there may be situations where COM forcibly running down (e.g. at process exit) can lead to issues with the DLL that implements held objects (think transitive, not just direct reference) being unloaded before CRT cleanup occurs, leading to dereferencing unloaded code.

Reply:
Filed a bug to address this issue: https://github.com/microsoft/WindowsAppSDK/issues/2183